### PR TITLE
Support pending

### PR DESCRIPTION
**Pull request by classicalliu**

### CODE REVIEW

File: packages/api-server/src/filter-web3-tx.ts [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+import { envConfig } from "./base/env-config";
+import { EthTransaction, EthTransactionReceipt } from "./base/types/api";
+import { Uint128, Uint32, Uint64 } from "./base/types/uint";
+import { PolyjuiceSystemLog, PolyjuiceUserLog } from "./base/types/gw-log";
+
+const CKB_SUDT_ACCOUNT_ID: U32 = 1;
+
+const EMPTY_ADDRESS = "0x" + "00".repeat(20);
+
+const GW_LOG_POLYJUICE_SYSTEM = "0x2";
```
Comment (RetricSu) [inline packages/api-server/src/filter-web3-tx.ts:23]: can we move these constant var to constant.ts file?

File: packages/api-server/src/filter-web3-tx.ts [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+    isCreate,
+    gasLimit,
+    gasPrice,
+    value,
+    inputSize: inputSize.toHex(),
+    input,
+  };
+}
+
+function accountIdToEthAddress(accountScriptHash: Hash, id: U32): HexString {
```
Comment (RetricSu) [inline packages/api-server/src/filter-web3-tx.ts:279]: do we still use this method? isn't contract address  = short_address = script_hash.slice(0, 42) ?

File: packages/api-server/src/filter-web3-tx.ts [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+  const fromScriptHash: Hash | undefined = await rpc.getScriptHash(fromId);
+  if (fromScriptHash == null) {
+    return undefined;
+  }
+  const fromScript: Script | undefined = await rpc.getScript(fromScriptHash);
+  if (fromScript == null) {
+    return undefined;
+  }
+
+  if (fromScript.code_hash !== envConfig.ethAccountLockHash) {
```
Comment (RetricSu) [inline packages/api-server/src/filter-web3-tx.ts:51]: will it be ok if we don't add this check? right now we have tron lock, and the web3-indexer include tx both for eth lock and tron lock (to support calling a eth-tx using tron wallet for signing)

Comment (RetricSu) [reply packages/api-server/src/filter-web3-tx.ts:51]: or add check for tron lock too ?
```suggestion
  if (fromScript.code_hash !== envConfig.ethAccountLockHash && fromScript.code_hash !== envConfig.tronAccountLockHash) {
```

File: packages/api-server/src/filter-web3-tx.ts (hunk 1 above)
Comment (classicalliu) [reply packages/api-server/src/filter-web3-tx.ts]: updated.

File: packages/api-server/src/filter-web3-tx.ts (hunk 2 above)
Comment (classicalliu) [reply packages/api-server/src/filter-web3-tx.ts]: updated.

File: packages/api-server/src/filter-web3-tx.ts (hunk 3 above)
Comment (classicalliu) [reply packages/api-server/src/filter-web3-tx.ts:50]: updated.

**Review by RetricSu — APPROVED** [review]

**Pull request merged**